File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.cpp
#include <bits/stdc++.h>

using namespace std;

int ret[1000004];
stack<int> stk;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    
    fill(ret, ret + 1000004, -1);
    
    for (int i=0; i<numbers.size(); i++){
        while(!stk.empty() && numbers[stk.top()] < numbers[i]){
            ret[stk.top()] = numbers[i];
            stk.pop();
        }
        stk.push(i);
    }
    
    for (int i=0; i<numbers.size(); i++){
        answer.push_back(ret[i]);
    }
    
    
    return answer;
}

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
function solution(numbers) {
    const answer = Array(numbers.length).fill(-1);
    const stk = [];
    
    numbers.forEach((e, i) => {
        while(stk.length > 0 && numbers[stk[stk.length - 1]] < e){
           answer[stk.pop()] = e;
        }
        stk.push(i);
    })
    
    return answer;
}